python3.9-pikepdf-2.12.2 fails to build on unstable-small

### Describe the bug
```console
building '/nix/store/vx0xf1j59m829ml37izwadrz130ziahg-python3.9-pikepdf-2.12.2.drv'...
builder for '/nix/store/vx0xf1j59m829ml37izwadrz130ziahg-python3.9-pikepdf-2.12.2.drv' failed with exit code 1; last 10 log lines:
  >       assert Name('/xyz') == b'/xyz'
  E       assert pikepdf.Name("/xyz") == b'/xyz'
  E        +  where pikepdf.Name("/xyz") = Name('/xyz')
  
  tests/test_object.py:196: AssertionError
  =========================== short test summary info ============================
  FAILED tests/test_object.py::test_unicode_fails - UnicodeEncodeError: 'utf-8'...
  FAILED tests/test_object.py::TestName::test_name_equality - assert pikepdf.Na...
  FAILED tests/test_object.py::test_bytes - assert pikepdf.Name("/xyz") == b'/xyz'
  ============= 3 failed, 440 passed, 8 skipped, 1 xfailed in 7.70s ==============
cannot build derivation '/nix/store/xx89191099n4h192jbx0cy1ajhff967s-python3.9-img2pdf-0.4.1.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/389048kx1gwb6jyi8h50w431da5q27c4-system-path.drv': 1 dependencies couldn't be built
```

### Steps To Reproduce
Steps to reproduce the behavior:
Try to install or build the derivation.

### Expected behavior
Successful update.

### Additional context
$ sudo nixos-rebuild boot -Q --upgrade

### Notify maintainers
@Kiwi 

### Metadata
```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.52, NixOS, 21.11pre304909.d0044b0e7d5 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.14`
 - channels(root): `"nixos-21.11pre305784.151c2f5a20c"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/development/python-modules/pikepdf/default.nix
+++ b/pkgs/development/python-modules/pikepdf/default.nix
@@ -24,12 +24,12 @@
 
 buildPythonPackage rec {
   pname = "pikepdf";
-  version = "2.12.2";
+  version = "2.16.1";
   disabled = ! isPy3k;
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "5ff35499b1ae7b181277f78ce5b1bcc8d3009182bb389917791c5dc811fcc8e4";
+    sha256 = "sha256-4k3/avMfHrcy/LXbRniDXR8xJkOZb9zZ2+uKylK8Dd4=";
   };
 
   buildInputs = [

